[refactor] Remove redundant queries of qconfig in `insertObservers`

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #31408 [quant][graphmode] `insert_quant_dequant` pass support shared class types
* #31407 [quant][graphmode][refactor] Remove observers in the end
* #31406 [quant][graphmode][refactor] Call `getQSchemeAndQParamMap` later in `quantizeTensors`
* #31293 [quant][graphmode][refactor] Get QScheme from observer module
* **#31292 [refactor] Remove redundant queries of qconfig in `insertObservers`**

Summary:
att
Also we need to do this check after we call `insertObservers` on invoked modules
as well since qconfig can be None for parent module while being valid for invoked modules

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D19146668](https://our.internmc.facebook.com/intern/diff/D19146668)

jerryzh168 committed:
[refactor] Remove redundant queries of qconfig in `insertObservers`

Summary:
att
Also we need to do this check after we call `insertObservers` on invoked modules
as well since qconfig can be None for parent module while being valid for invoked modules

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kostmo commented:
## CircleCI build failures summary
As of commit 64f7c85:
* **2/2** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=64f7c858c1bba1db5063da1f10ea4387725bd87e).

### 2 failures *not* recognized by patterns:
| Job | Step | Status |
| --- | --- | --- |
| ![CircleCI](https://avatars0.githubusercontent.com/ml/7?s=12) <sup>caffe2_onnx_py2_gcc5_ubuntu16_04_test</sup> | <sup>Test</sup> | New in PR |
| ![CircleCI](https://avatars0.githubusercontent.com/ml/7?s=12) <sup>caffe2_onnx_py3_6_clang7_ubuntu16_04_test</sup> | <sup>Test</sup> | New in PR |


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

jerryzh168 committed:
Update on "[refactor] Remove redundant queries of qconfig in `insertObservers`"


Summary:
att
Also we need to do this check after we call `insertObservers` on invoked modules
as well since qconfig can be None for parent module while being valid for invoked modules

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Remove redundant queries of qconfig in `insertObservers`"


Summary:
att
Also we need to do this check after we call `insertObservers` on invoked modules
as well since qconfig can be None for parent module while being valid for invoked modules

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Remove redundant queries of qconfig in `insertObservers`"


Summary:
att
Also we need to do this check after we call `insertObservers` on invoked modules
as well since qconfig can be None for parent module while being valid for invoked modules

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -458,16 +458,20 @@ void InsertObserversHelper::addIntermediateValuesToSkipObserver(
     insertObservers(invoked_module, invoked_method_name);
   }
 
+  auto qconfig_opt = module_qconfig_map_.at(module._ivalue());
```
Please add a comment explaining why this needs to happen after iterating through invoked methods (the one you have in the commit message is good).

ZolotukhinM reviewed (approved):
👍

jerryzh168 committed:
Update on "[refactor] Remove redundant queries of qconfig in `insertObservers`"


Summary:
att
Also we need to do this check after we call `insertObservers` on invoked modules
as well since qconfig can be None for parent module while being valid for invoked modules

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D19146668](https://our.internmc.facebook.com/intern/diff/D19146668)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Remove redundant queries of qconfig in `insertObservers`"


Summary:
att
Also we need to do this check after we call `insertObservers` on invoked modules
as well since qconfig can be None for parent module while being valid for invoked modules

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D19146668](https://our.internmc.facebook.com/intern/diff/D19146668)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Remove redundant queries of qconfig in `insertObservers`"


Summary:
att
Also we need to do this check after we call `insertObservers` on invoked modules
as well since qconfig can be None for parent module while being valid for invoked modules

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D19146668](https://our.internmc.facebook.com/intern/diff/D19146668)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Remove redundant queries of qconfig in `insertObservers`"


Summary:
att
Also we need to do this check after we call `insertObservers` on invoked modules
as well since qconfig can be None for parent module while being valid for invoked modules

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D19146668](https://our.internmc.facebook.com/intern/diff/D19146668)

[ghstack-poisoned]
